### ISSUE TITLE
Pyright report false positive for generator function annotated with `Iterator[T]`

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#1289]
Pyright report false positive for generator function annotated with `Iterator[T]` as return type. Examples from typing module document:

```python
def infinite_stream(start: int) -> Iterator[int]:
    while True:
        yield start
        start += 1
```

Pyright gives error on line `yield start`:

```
Expression of type "Iterator[int]" cannot be assigned to yield type "Iterable[int]"
  "__class_getitem__" is an incompatible type
    Type "(cls: Type[Iterator[int]], item: Any) -> GenericAlias" cannot be assigned to type "(cls: Type[Iterable[TypeVar('_T_co')]], item: Any) -> GenericAlias"
      Parameter 2: type "Type[Iterable[TypeVar('_T_co')]]" cannot be assigned to type "Type[Iterator[int]]"
        "__next__" is not present
        Type "Type[Iterable[TypeVar('_T_co')]]" cannot be assigned to type "Type[Iterator[int]]"
```

**VS Code extension or command-line**
VS Code extension v1.1.95

**Comment by erictraut** [comment]
I'm not able to repro the problem with the sample above. Perhaps there's additional code involved? Maybe a class that has a `__class_getitem__` method defined?

**Comment by yaxum62** [comment]
I can produce it with a single file without dependencies:

![issue](https://user-images.githubusercontent.com/8572778/102575827-f7bd6500-40a8-11eb-80a6-9da3f503d03a.jpg)

Not sure if python version could be involved, but I'm using python 3.9.1

**Comment by erictraut** [comment]
Hmm, very strange. I'm not able to repro, and I can't explain the message you are seeing.

Are you using custom typeshed stubs for (e.g. for typing.pyi), or are you using the default stubs that are packaged with pyright?

**Comment by yaxum62** [comment]
I don't think so, this is my pyrightconfig.json:

```json
{
  "venvPath": ".",
  "venv": ".venv"
}
```

and pyright server log:

```
[Info  - 9:06:02 PM] Pyright language server 1.1.95 starting
[Info  - 9:06:02 PM] Server root directory: /home/yaxum/.vscode-server/extensions/ms-pyright.pyright-1.1.95/dist
[Info  - 9:06:02 PM] Background analysis(1) root directory: /home/yaxum/.vscode-server/extensions/ms-pyright.pyright-1.1.95/dist
[Info  - 9:06:02 PM] Background analysis(1) started
Waiting for Python extension to load
Python extension loaded
Received pythonPath from Python extension: python
[Error - 9:06:03 PM] stubPath /src/foo/typings is not a valid directory.
```

I tried created this file in a different project, open this file with no project, reboot my computer, reboot wsl, uninstall and reinstall pyright plugin, but the issue is still there.

However, that issue is gone when I opened that file on my windows host. Do you think this is a WSL related issue?

**Comment by yaxum62** [comment]
Actually I take that back, I found I changed typeshed settings from vscode preference.

Sorry for the buzz, happy holidays!
